src/sentry_core.c

```diff
@@ -740,24 +739,36 @@ sentry_transaction_start(sentry_value_t tx_cxt)
             sentry__msec_time_to_iso8601(sentry__msec_time())));
 
     sentry_value_decref(tx_cxt);
-
     return tx;
 }
 
 sentry_uuid_t
 sentry_transaction_finish(sentry_value_t tx)
 {
-    // The sampling decision should already be made for transactions during
-    // their construction. No need to recalculate here. See
+    if (sentry_value_get_type(tx) == SENTRY_VALUE_TYPE_STRING) {
+        sentry_value_decref(tx);
+        tx = sentry_value_new_null();
+        // Minimize the amount of time spent locking the scope, and do checking
+        // on the transaction when this is not holding onto the lock.
+        SENTRY_WITH_SCOPE (scope) {
+            tx = sentry__value_clone(scope->span);
```

Comment:
I think I commented in another PR that you can just move this, instead of copy+delete.